application/src/main/java/org/togetherjava/tjbot/commands/system/BotCore.java

```diff
@@ -268,4 +281,22 @@ private static void onComponentIdRemoved(ComponentId componentId) {
          */
         void accept(A first, B second, C third);
     }
+
+    private static final class MessageReceiverAsEventListener extends ListenerAdapter {
+        private final MessageReceiver messageReceiver;
+
+        MessageReceiverAsEventListener(MessageReceiver messageReceiver) {
+            this.messageReceiver = messageReceiver;
+        }
+
+        @Override
+        public void onGuildMessageReceived(@NotNull GuildMessageReceivedEvent event) {
+            messageReceiver.onMessageSent(event);
+        }
+
+        @Override
+        public void onGuildMessageUpdate(@NotNull GuildMessageUpdateEvent event) {
+            messageReceiver.onMessageUpdated(event);
+        }
```

Comment:
I'd personally we say, we implement EventListener and do instanceof checks ourselves.

Currently it goes from onEvent -> onGuildMessageUpdate -> onMessageUpdated, I don't think we need the ListenerAdapter

Reply:
Sounds reasonable.

Reply:
done